pkg/storkctl/clusterdomainupdate_test.go

```diff
@@ -0,0 +1,223 @@
+	require.Equal(t, name, cdu.Name, "ClusterDomainUpdate name mismatch")
+	require.Equal(t, cdu.Spec.ClusterDomain, clusterDomain, "ClusterDomain name mismatch")
+	require.Equal(t, cdu.Spec.Active, active, "Active value mismatch")
+	return cdu
+}
+
+func TestGetClusterDomainUpdateNoUpdates(t *testing.T) {
+	cmdArgs := []string{"get", "clusterdomainupdate"}
+
+	expected := "No resources found.\n"
+	testCommon(t, cmdArgs, nil, expected, false)
+}
+
+func TestGetClusterDomainUpdateShortNoUpdates(t *testing.T) {
+	cmdArgs := []string{"get", "cdu"}
+
+	expected := "No resources found.\n"
+	testCommon(t, cmdArgs, nil, expected, false)
+}
+
+func TestGetClusterDomainUpdate(t *testing.T) {
+	defer resetTest()
+	createClusterDomainUpdate(t, "test1", "zone1", true)
+	cmdArgs := []string{"get", "clusterdomainupdate", "test1"}
+
+	expected := "NAME      CLUSTER DOMAIN   UPDATE ACTION   UPDATE STATUS   CREATED\n" +
+		"test1     zone1            Activate                        \n"
+	testCommon(t, cmdArgs, nil, expected, false)
+}
+
+func TestGetClusterDomainUpdateWithChanges(t *testing.T) {
+	defer resetTest()
+	cdu := createClusterDomainUpdate(t, "test1", "zone1", true)
+	cmdArgs := []string{"get", "clusterdomainupdate", "test1"}
+
+	expected := "NAME      CLUSTER DOMAIN   UPDATE ACTION   UPDATE STATUS   CREATED\n" +
+		"test1     zone1            Activate                        \n"
+	testCommon(t, cmdArgs, nil, expected, false)
+
+	cdu.Status.Status = storkv1.ClusterDomainUpdateStatusFailed
+	_, err := k8s.Instance().UpdateClusterDomainUpdate(cdu)
+	require.NoError(t, err, "Error updating cluster domain")
+
+	cmdArgs = []string{"get", "clusterdomainupdate", "test1"}
+
+	expected = "NAME      CLUSTER DOMAIN   UPDATE ACTION   UPDATE STATUS   CREATED\n" +
+		"test1     zone1            Activate        Failed          \n"
+	testCommon(t, cmdArgs, nil, expected, false)
+
+	cdu.Status.Status = storkv1.ClusterDomainUpdateStatusSuccessful
+	_, err = k8s.Instance().UpdateClusterDomainUpdate(cdu)
+	require.NoError(t, err, "Error updating cluster domain")
+
+	cmdArgs = []string{"get", "clusterdomainupdate", "test1"}
+
+	expected = "NAME      CLUSTER DOMAIN   UPDATE ACTION   UPDATE STATUS   CREATED" +
+		"\ntest1     zone1            Activate        Successful      \n"
+	testCommon(t, cmdArgs, nil, expected, false)
+
+}
+
+func TestGetClusterDomainUpdateNotFound(t *testing.T) {
+	defer resetTest()
+	createClusterDomainUpdate(t, "test1", "zone1", true)
+	cmdArgs := []string{"get", "clusterdomainupdate", "test2"}
+
+	expected := `Error from server (NotFound): clusterdomainupdates.stork.libopenstorage.org "test2" not found`
+	testCommon(t, cmdArgs, nil, expected, true)
+}
+
+func TestActivateClusterDomain(t *testing.T) {
+	defer resetTest()
+	name := "zone2"
+	cmdArgs := []string{"activate", "clusterdomain", name}
+
+	expected := ""
+	testCommon(t, cmdArgs, nil, expected, false)
+
+	cdus, err := k8s.Instance().ListClusterDomainUpdates("kube-system")
```

Comment:
Call `storkctl get` like below to verify

Reply:
If a name is not provided to `storkctl activate` command, storkctl will generate a uuid for this cluster domain update. There is no way I can check if `storkctl get` output matches.

That is why I added similar tests `WithName` where a name is provided like `storkctl activate clustrdomain <zone1> --name update-zone1`, and then `storkctl get` is used to test the output.